### PR TITLE
Single page app

### PR DESCRIPTION
**Pull request by Naveen-g09**
- [x] create a single page only for tasks
- [x] add feature to drag to refresh the api call for tasks
- [x] add a time to refresh the api call for tasks (preferably every 5 minutes)
- [x] tasks will be fetched from api
- [x] table format is: name, number, city(optional), state(optional), target call count, created at(by default), modified at
- [x] call-logs should be configured to local database expo sqlite
- [x] call log databse table : Timestamp [Event at: ], Task ID [foreign key from api table], Call-Type [Event type] {a drop down with options outgoing, incoming, missed, unable to connect}, Duration of call

### Tech stack 

> Drizzle
> Expo-Sqlite

##Process

@sutyum sir update this, so it will be clear to me and will help me to learn

### CODE REVIEW

**Comment by sutyum** [comment]
- [ ] Use react query and drizzle to interface with sqlite instances
- [x] Add expo-sqlite and turso to interface through Drizzle
- [ ] add function to sync CallLogs from mobile to cloud
- [ ] add function to sync CallTasks from cloud to mobile
- [ ] Fetch call logs from phone prepare to push to db
- [x] Trigger both syncs on refresh
- [x] add pull on refresh to the list

**Comment by Naveen-g09** [comment]
@sutyum sir add comments in the code, wherever you feel something is lacking, will fix it first thing in morning

File: app/(tabs)/_layout.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          tabBarIcon: ({ color }) => <TabBarIcon name="code" color={color} />,
+          headerRight: () => (
+            <Link href="/inbox" asChild>
+              <Pressable>
+                {({ pressed }) => (
+                  <FontAwesome
+                    name="info-circle"
+                    size={25}
+                    color={Colors[colorScheme ?? "light"].text}
+                    className={`mr-15 ${
```
Comment (sutyum) [inline app/(tabs)/_layout.tsx:40]: https://github.com/dcastil/tailwind-merge

**Comment by sutyum** [comment]
> @sutyum sir add comments in the code, wherever you feel something is lacking, will fix it first thing in morning

is it working?

File: anything.ts
Comment (sutyum) [inline anything.ts:1]: ?

File: app/[...missing].tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ export default function NotFoundScreen() {
       <View className="flex-1 items-center justify-center p-5">
         <Text className="text-2xl font-bold">This screen doesn't exist.</Text>
 
-        <Link href="/index" className="mt-4 py-4">
+        <Link href="/" className="mt-15 py-15">
```
Comment (sutyum) [inline app/[...missing].tsx:13]: This sort of styling will break on various screens, consider using auto margins instead

File: database.ts
Comment (sutyum) [inline database.ts:1]: pending deletion

File: drizzle/index.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { createClient } from "@libsql/client";
+import dotenv from "dotenv";
+import { drizzle as expoDrizzle } from "drizzle-orm/expo-sqlite";
+import { drizzle as tursoDrizzle } from "drizzle-orm/libsql";
+import { openDatabaseSync } from "expo-sqlite/next";
+import path from "path";
+
+import { callLogs, tasks } from "./schema";
```
Comment (sutyum) [inline drizzle/index.ts:9]: 'tasks' is defined but never used

File: lib/client.ts
Comment (sutyum) [inline lib/client.ts:1]: looks good

File: metro.config.js [hunk 4]
```diff
@@ -6,6 +6,9 @@ const config = getDefaultConfig(__dirname, {
   // [Web-only]: Enables CSS support in Metro.
   isCSSEnabled: true,
 });
+
+config.resolver.sourceExts.push("sql");
```
Comment (sutyum) [inline metro.config.js:10]: ?

File: react-native-call-log.d.ts
Comment (sutyum) [inline react-native-call-log.d.ts:1]: nice

**Review by sutyum — CHANGES_REQUESTED** [review]
Your code is looking fine for what is needed here, as long as it works. I have avoided adding unnecessary improvements that you can make as we need to ship this fast and move to ella

**Comment by Naveen-g09** [comment]
> > @sutyum sir add comments in the code, wherever you feel something is lacking, will fix it first thing in morning
> 
> is it working?

Nope, we figured out way to configure dual databases, and called sync function 

>some part of it is left
Migrations are working 
So hopefully it will be done in the first half

File: anything.ts
Comment (Naveen-g09) [reply anything.ts:1]: me and prince were testing the db, will remove in next commit
it didn't work.

Comment (Naveen-g09) [reply anything.ts:1]: deleted it

**Comment by Naveen-g09** [comment]
@sutyum sir please guide on work ahead
I believe zod implementation on schema is left and fetch and get request need to be refactored
currently it is a mess, If it is a 2 hours work, I will be do it tommorrow
Please guide me on how to resolve android bundling issue
image

**Pull request closed without merging**